src/js/components/DataTable/Header.js

```diff
@@ -88,6 +88,28 @@ const StyledContentBox = styled(Box)`
   ${props => props.extend}
 `;
 
+const calcBackground = (backgroundProp, pin, theme) => {
```

Comment:
Any chance we could extract this to a reusable function that also takes `tableContext` as a parameter? then you could say `theme.dataTable.pinned[tableContext]` and re-use across Footer also

Reply:
I thought about it although it seems a little odd and obfuscated. Seemed like I had to go look at the function to figure out what it actually did if I had that many args

Reply:
Done